raft-tech/TANF-app
# Feat/#1814 #1840 #1839 #1334 account access emails

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1814 #1840 #1839 #1334

Develop demo:
https://gorafttech-my.sharepoint.com/personal/msohani_goraft_tech/_layouts/15/stream.aspx?id=%2Fpersonal%2Fmsohani%5Fgoraft%5Ftech%2FDocuments%2FRecordings%2FOffice%20hours%5Fopen%20pairing%2D20221012%5F130511%2DMeeting%20Recording%2Emp4&ga=1

~~I had to merge in these commits from develop:
https://github.com/raft-tech/TANF-app/pull/2074/files#diff-2dccf6d7d74fa8f11c6e941c96845865d8ce23de87ce36e8990a2bc9adf39da3~~
~~They should have been merged into [release/v2.1.0](https://github.com/raft-tech/TANF-app/tree/release/v2.1.0).
There were probably other commits I missed because I still needed to make some front end changes. As such, there are additional commits on this branch that are not directly related to the emails. I believe I did this correctly, but all the file changes should be checked.~~

Merged `develop` into `release-2.1.0`, updated branch

## How to Test
_List the steps to test the PR_

#1814
1. Use label to deploy to a dev env or deploy locally.
2. Set account_approval_status set to Initial.
3. Login and request access with form and button.
4. An account requested email should be sent if deployed on dev or in the console if deployed locally


#1840
1. Use label to deploy to a dev env or deploy locally.
2. Login and go to Admin page.
3. Go to user and set account_approval_status to Deactivated.
4. An account deactivated email should be sent if deployed on dev or in the console if deployed locally

#1839
1. Use label to deploy to a dev env or deploy locally.
2. Login and go to Admin page.
3. Go to user and set account_approval_status to Approved.
4. An account approved email should be sent if deployed on dev or in the console if deployed locally

#1334
1. Use label to deploy to a dev env or deploy locally.
2. Login and go to Admin page.
3. Go to user and set account_approval_status to Denied.
4. An account denied email should be sent if deployed on dev or in the console if deployed locally


## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._


#1814
- [x] When  `account_approval_status` field changes to `access request`, this should trigger a `Access request` email to the user that confirms that their request for access has been submitted.
- [x] No other changes applied to `account_approval_status`  field should trigger a `Access request` email to the user. 
- [x] Testing Checklist and confirm all tests pass

#1840
- [x] When an `OFA System Admin` changes a user's `account_approval_status` status to `Deactivated` in the user's profile, this will trigger a `deactivated` email to the user. 
- [x] No other status changes to `account_approval_status`  will trigger a `deactivated` email to the user. 

#1839
- [x] When an `OFA System Admin` changes a user's `account_approval_status` status to `Approved` in the user's profile, this will trigger an `approved` email to the user. 
- [x] No other status changes to `account_approval_status`  will trigger a `approved` email to the user. 

#1334
- [x] When an `OFA System Admin` changes a user's `account_approval_status` status to `Denied` in the user's profile, this will trigger a `deny` email to the user. 
- [x] No other status changes to `account_approval_status`  will trigger a `deny` email to the user. 


### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)


_email templates approved by @ttran-hub [here](https://github.com/raft-tech/TANF-app/issues/1708#issuecomment-1183521341)_

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

_n/a_
_n/a_

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

_n/a_
